src/main/java/com/firebolt/jdbc/service/FireboltAuthenticationService.java

```diff
@@ -51,7 +53,11 @@ public FireboltConnectionTokens getConnectionTokens(String host, FireboltPropert
 		} catch (FireboltException e) {
 			log.log(Level.SEVERE, "Failed to connect to Firebolt", e);
 			String msg = ofNullable(e.getErrorMessageFromServer()).map(m -> format(ERROR_MESSAGE_FROM_SERVER, m)).orElse(format(ERROR_MESSAGE, e.getMessage()));
-			throw new FireboltException(msg, e);
+			SQLState sqlState = SQLState.fromCode(e.getSQLState());
+			if (sqlState == null) {
```

Comment:
nit: Do we need this if? Can we just put whatever value we get as a code?

Reply:
You're right. I had an issue with null value before so I had to add this check. However, with a fresh mind today I've found an arguably neater solution.